pkg/pgn/parser.go

```diff
@@ -15,6 +16,43 @@ import (
 	"github.com/inhies/go-bytesize"
 )
 
+func PathParser(pgn_path string) ([]string, error) {
+	var files []string
+	fileTypes := []string{"pgn", "zst", "bz2"}
+	paths := strings.Split(pgn_path, ",")
+
+	for _, path := range paths {
+		file, err := os.Open(path)
+		if err != nil {
+			return nil, fmt.Errorf("error opening PGN: %s", err)
+		}
+		fileInfo, err := file.Stat()
+		if err != nil {
+			return nil, fmt.Errorf("error getting file stats: %s", err)
+		}
+
+		if fileInfo.IsDir() {
+			subfiles, err := ioutil.ReadDir(path)
```

Comment:
if you read the description from ioutil.ReadDir() docs you can see it is deprecated.
deprecated functions should still work but in general should be avoided.
if feasible replace with os.ReadDir() as suggested in docs